### ISSUE TITLE
exploit/multi/script/web_delivery PSH target /w SSL option set to true, results in PSH stager syntax error.

### ISSUE THREAD

**Issue by Bruk0ut** [rapid7/metasploit-framework#7309]
## Steps to reproduce

TL;DR, If you use the exploit/multi/script/web_delivery module, specify the PSH target and set SSL to true, the framework will generate a PSH stager with a syntax error because it is missing a semi-colon after the "[Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}" call.

How'd you do it?

1. Use exploit/multi/web_delivery module
2. Set SSL to true
3. Specify a payload (I used windows/meterpreter/reverse_https) /w required options (LHOST,LPORT etc)
4. Run the module
5. Copy the PSH stager code that it generates
6. On a Windows box /w PowerShell installed (I tested with /w Win 8.1 64 bit Enterprise), open a cmd prompt and paste the PSH stager code, but remove the "-w hidden" argument so that you can see the error that PowerShell will generate).
7. Execute the PSH stager from cmd prompt and observe the error.

It should be a quick fix by just having the framework insert a semi-colon after the "[Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}" call.

Manually placing a semi-colon resolves this issue and the PSH stager is executed.

## Expected behavior
PSH Stager should execute and the payload ultimately should execute

## Current behavior
PowerShell errors with syntax error

## System stuff
Framework running on Kali 2016.1 rolling release, 64bit
Framework: 4.12.23-dev
Console: 4.12.23-dev

Windows target running Windows 8.1 Enterprise, 64bit

### I installed Metasploit with:
- [x ] Kali package via apt

### OS
Framework running on Kali 2016.1 rolling release, 64bit

See msf_error_1.png screenshot showing all options and the "bad" generated PSH stager after running the module. Area highlighted where the semi-colon is missing.

See msf_error_2.png screenshot showing the PowerShell error when run manually from the Windows client (with "-w hidden" removed to show the error). Area highlighted in yellow where the semi-colon is missing.

![msf_error_1](https://cloud.githubusercontent.com/assets/5972704/18497521/de0d0e52-79e3-11e6-9a5d-90f0684d4879.png)

![msf_error_2](https://cloud.githubusercontent.com/assets/5972704/18497526/e6d306e0-79e3-11e6-90b9-cd6cdd308a4a.png)

Cheers

--Bruk0ut

EDIT: Updated with the full correct module path (I accidentally omitted "exploit/multi/**script**/web_delivery in the original post).

**Comment by bwatters-r7** [comment]
Thanks for the effort to help us out on this, @Bruk0ut!

**Comment by dmaloney-r7** [comment]
@Bruk0ut https://github.com/rapid7/metasploit-framework/pull/7374 fixes the specific issue you mentioned but we still can't seem to get it to work. it sounds from your ticket like it did work for you. are we missing something when we try to repro?

**Comment by dmaloney-r7** [comment]
@Bruk0ut nm, it was a stupid environmental issue

**Comment by pbarry-r7** [comment]
Verified as fixed via [PR-7374](https://github.com/rapid7/metasploit-framework/pull/7374).
